ImageData/FileLoader.h

```diff
@@ -136,10 +136,16 @@ class FileLoader {
     // Return a casacore image type representing the data stored in the
     // specified HDU/group/table/etc.
     virtual ImageRef LoadData(FileInfo::Data ds) = 0;
-    virtual bool GetCursorSpectralData(std::vector<float>& data, int stokes, int cursor_x, int cursor_y);
-    virtual std::map<CARTA::StatsType, std::vector<double>>* GetRegionSpectralData(
-        int stokes, int region_id, const casacore::ArrayLattice<casacore::Bool>* mask, IPos origin);
+    virtual bool GetCursorSpectralData(std::vector<float>& data, int stokes, int cursor_x, int count_x, int cursor_y, int count_y);
+    // check if one can apply swizzled data under such image format and region condition
+    virtual bool CanUseSiwzzledData(const casacore::ArrayLattice<casacore::Bool>* mask);
```

Comment:
Updated comment: typo is still a typo; but if this is on the loader now I would suggest a different name -- `HasRegionSpectralData`.

Reply:
Actually, `UseRegionSpectralData` is probably better -- it may *have* the data, but this function determines whether to *use* it or not.